go-client/client/tablehandle.go

```diff
@@ -10,9 +11,18 @@ import (
 	ticketpb2 "github.com/deephaven/deephaven-core/go-client/internal/proto/ticket"
 )
 
+// ErrInvalidTableHandle is returned by most table methods
+// when called on a table handle that contains its zero value or has been already released.
+var ErrInvalidTableHandle = errors.New("tried to use a zero-value or released table handle")
+
 // A TableHandle is a reference to a table stored on the deephaven server.
 //
-// It should be eventually released using `Release()` once it is no longer needed.
+// It should eventually be released using Release() once it is no longer needed on the client.
+// Releasing a table handle does not affect table handles derived from it.
+//
+// All TableHandle methods (with the exception of Release) are goroutine-safe.
```

Comment:
`Release` does not actually change the client. It defers immediately to `sessionStub.release`, which also does not change the sessionStub. The only reason I marked `Release` as not being goroutine safe is because it could release a table that is currently being used by another method.